src/Momento.Sdk/Auth/ICredentialProvider.cs

```diff
@@ -2,9 +2,21 @@ namespace Momento.Sdk.Auth;
 
 using System.Collections.Generic;
 
+/// <summary>
+/// Interface for credentials
```

Comment:
Best practice for documentation strings:

If your docstring uses the same words as the definition/name, then it's not really adding any value.  So e.g. you don't need to say `interface` here, that will already be visible to the user.

Maybe say something like:

Specifies the fields that are required for a Momento client to connect to and authenticate with the Momento service.